pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -82,11 +83,18 @@ func NewMachineScope(params MachineScopeParams) (*MachineScope, error) {
 		return nil, fmt.Errorf("failed to get machine provider status: %w", err)
 	}
 
-	cloudEnv, armEndpoint, err := GetCloudEnvironment(params.CoreClient)
+	infraTags, infraName, cloudEnv, armEndpoint, err := GetCloudEnvironment(params.CoreClient)
 	if err != nil {
 		return nil, fmt.Errorf("failed azure environment: %w", err)
 	}
 
+	tags, err := getTagList(map[string]string{
+		fmt.Sprintf("kubernetes.io_cluster.%v", infraName): "owned",
+	}, infraTags, machineConfig.Tags)
```

Comment:
Can we make whatever this first map is into a function that takes the infraName and returns this map, so we can give the function a name that describes what this is for